### PR TITLE
ipynb: Create a notebook for lava processing

### PR DESCRIPTION
**Pull request by vingu-linaro**
Create a dedicated notebokk for lava postprocessing that can be used
either in lava job to post process WA results with its .py version
or as a normal python notebook to visualize local results

The .py file will generate .csv file with results that will be then
uploaded in qa-report

Signed-off-by: Vincent Guittot <vincent.guittot@linaro.org>

### CODE REVIEW

File: ipynb/wltests/lava-post-processing.py [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+# ## Results analysis and metrics collection
+
+# In[ ]:
+
+
+collector = WaResultsCollector(
+    
+    # WLTests results folder:
+    base_dir='/root/output/', # Base path of your results folders
+    wa_dirs='wa',   # Parse only folder matching this regexp
```
[the remark this answers is not in the document]
Comment (vingu-linaro) [reply ipynb/wltests/lava-post-processing.py]: I have to check what can be done in the lava job to parametrize this. But the goal was to just run this script and avoid thousand of parameters in the lava job that already got a significant number

**Comment by vingu-linaro** [comment]
> **Regarding the notebook:**
> 
> I'd like to avoid having another wltests notebook that duplicates most of https://github.com/ARM-software/lisa/blob/next/ipynb/wltests/sched-evaluation-full.ipynb. I'd be fine with extending that notebook to cover more cases however.

I'm fine to keep only one .ipynb. In fact, that's what we have currently. I have created  this new ipynb to for the pull request and make it easier to merge :-)

> 
> **Regarding the script:**
> 
> It should be moved to where other scripts are (`$repo/tools/scripts`). Also, it would be nice to clean out the hunks left by the .ipynb -> .py conversion.

Would it be easier to maintain the .py if we keep both  .py and .ipynb in the same location ? This will help to make sure they are always synced

What do you mean by clean out the hunks left by the .ipynb -> .py conversion ?

**Comment by valschneider** [comment]
> Would it be easier to maintain the .py if we keep both .py and .ipynb in the same location ? This will help to make sure they are always synced

Right, so thinking about it it would be nice if we could remove the duplication altogether by only having the `.ipynb` file and rely on automatic `.py` translation (https://nbconvert.readthedocs.io/en/latest/execute_api.html#executing-notebooks-from-the-command-line). It will be a pain if you had to tweak the converted `.py` file, but if we could get a `.ipynb` that can smoothly be translated to `.py` it would lessen the maintenance burden.

> What do you mean by clean out the hunks left by the .ipynb -> .py conversion ?

Mostly all of the extra whitespace and notebook cell remains (e.g. `# In[ ]:`).

**Comment by vingu-linaro** [comment]
> > Would it be easier to maintain the .py if we keep both .py and .ipynb in the same location ? This will help to make sure they are always synced
> 
> Right, so thinking about it it would be nice if we could remove the duplication altogether by only having the `.ipynb` file and rely on automatic `.py` translation (https://nbconvert.readthedocs.io/en/latest/execute_api.html#executing-notebooks-from-the-command-line). It will be a pain if you had to tweak the converted `.py` file, but if we could get a `.ipynb` that can smoothly be translated to `.py` it would lessen the maintenance burden.

At now, we do: jupyter nbconvert --to script [YOUR_NOTEBOOK].ipynb to create the py once and save it.  I don't want to add to much overhead in the execution of the lava job as this will have to be done for each job.
And i'm not sure that jupyter is installed in the lava job as we don't need it.

I will see what can be done without adding too much overhead. We already have around 4h of tests for jankbench and multimedia and around 2 h for vellamo

> 
> > What do you mean by clean out the hunks left by the .ipynb -> .py conversion ?
> 
> Mostly all of the extra whitespace and notebook cell remains (e.g. `# In[ ]:`).

IMHO, it makes it easier to read as we can easily compare it with the original .ipynb. Furthermore, this will be even more difficult with on the fly conversion proposed above

**Comment by valschneider** [comment]
> At now, we do: jupyter nbconvert --to script [YOUR_NOTEBOOK].ipynb to create the py once and save it. I don't want to add to much overhead in the execution of the lava job as this will have to be done for each job.
> And i'm not sure that jupyter is installed in the lava job as we don't need it.
> 
> I will see what can be done without adding too much overhead. We already have around 4h of tests for jankbench and multimedia and around 2 h for vellamo

If there's no extra modifications to be done on the converted `.py` script then I'd rather have just the `.ipynb` in LISA and perhaps add some comments explaining that the notebook can be easily run as a script for CI purposes (and how to do so).

> IMHO, it makes it easier to read as we can easily compare it with the original .ipynb. Furthermore, this will be even more difficult with on the fly conversion proposed above

Yeah that'd only be relevant if we do end up with both `.ipynb` and `.py`

**Pull request closed without merging**